hazelcast/src/main/java/com/hazelcast/internal/util/ServiceLoader.java

```diff
@@ -263,6 +263,9 @@ public boolean equals(Object o) {
             if (uri != null ? !uri.equals(that.uri) : that.uri != null) {
                 return false;
             }
+            if (classLoader != null ? !classLoader.equals(that.classLoader) : that.classLoader != null) {
```

Comment:
you can use `Objects.equals()`

Reply:
🤦  copy-paste from JDK6-compatible 3.12.z patch